### DISCUSSION TITLE
Multiple depot with limited vehicle

### DISCUSSION THREAD

**Discussion by KenAragorn** [google/or-tools#2710] [Routing (and legacy CP) questions]
Hi All,

We are exploring route planner for our delivery services. However, we have some questions and we seems not able to find a confirmation on this, which is:

Can we have a multiple depot in which has a pre-defined vehicles list. That means, this week we will have:

Week 1
- 3 Depots
- 5 vehicle riders for delivery
- 60 delivery points in which 20 from depot A, 30 depot B and 10 from depot C and each parcels is unique and available at the selected depot
- based on the 5 rider we have, system are expected to select the most minimum riders needed for delivery and each rider have a max capacity of 15 parcels and rider

My initial thoughts are to pre-prepared the data in which: 

1. Prepare the delivery points by depot. Which mean for depot A which got 20 parcels to be deliver out, based on our own business rules which is max 15 parcels per rider per day, we will then assign 2 riders and not 5. Thus, the final data preparation to OR-Tools is :

- 20 delivery points
- 2 riders
- 1 depot

....and from there, get the planned and recommended routing.

Then once the above job done, update the rider status to 'assigned'. We then repeat the above for depot 2 and 3 with the left over rider.

But, it seems quite tedious. I'm exploring whether is there any way OR-Tools provide such thing out of the box. Which mean, it can support multiple depot (OR-Tools is already supporting it) but also the 'constraint' for the delivery points each got their own starting points pre-defined.

Appreciate the helps and tips.

Thanks!

**Comment by lperron** [comment]
Do not expect to the solver to decompose the model for you, especially in this case. You have a simple schema. At the bare minimum, you will remove the need for the solver to do connected components and scan the whole dataset just to partition it.

**Reply by KenAragorn** [reply]
Ok, got it. So, decompose all the require model, and let the solver (ORTools) to focus only 1 job and for the special need in my case, if there are needs for multiple depot with a specific rules listed above, 'decompose' it beforehand and solver remain as 1 depot.
